physical/azure/azure.go

```diff
@@ -84,17 +84,29 @@ func NewAzureBackend(conf map[string]string, logger log.Logger) (physical.Backen
 	}
 
 	var environment azure.Environment
+	var URL *url.URL
 	var err error
 
-	if environmentURL != "" {
-		environment, err = azure.EnvironmentFromURL(environmentURL)
-		if err != nil {
-			return nil, fmt.Errorf("failed to look up Azure environment descriptor for URL %q: %w", environmentURL, err)
+	testHost := conf["testHost"]
```

Comment:
I wonder if there is a possibility of `testHost` being configured outside the test? Not sure if this is a valid concern though.

Reply:
Yes, this does modify prod code, so a Vault user could make use of this undocumented option.  I'm okay with that personally, but if you foresee a problem speak up.

Reply:
Perhaps we could rename `tesHost` to something that denotes that this configuration is not for prod. I would suggest something like `_testHost_` or  `_ciTestHost_`?

Reply:
```suggestion
	// used by integration tests
	testHost := conf["_testHost_"]
```

Reply:
Having the `test` prefix (and not documenting it) is good enough on that front for me.  There's plenty of examples of other such conf in our codebase, e.g. Raft's apply_delay: https://github.com/hashicorp/vault/blob/main/physical/raft/raft.go#L337

Reply:
Sounds good!